### PR TITLE
Write local testing config values to AMP runtime

### LINKED ISSUE #12054: Running local server on non-localhost causes breakages
Steps to reproduce:
```
gulp --host=<insert name of host machine>
```
navigate to hostname:8000/examples/ads.amp.html
Select doubleclick for ad network.
Notice how all the ads 404

The issue is that non-localhost is not being identified as localDev in mode.js, which is instead hardcoding localhost in its check for localdev mode.

### PR DESCRIPTION
**Pull request by rsimha-amp**
This PR does the following:
1. Makes sure that the values of `localDev: true`, `thirdPartyFrameHost`, and `thirdPartyFrameRegex` are written to the local AMP config to enable local testing with `gulp build` and `gulp dist --fortesting`.
2. Adds a new parameter `--local_dev` to `gulp prepend-global`, which will ensure that the AMP runtime is set up for local development and testing. 

With this, `localDev` will be set to true when the runtime is built and served on `localhost` via `gulp`, or via the following commands:
```
gulp clean
gulp build
gulp serve
```

In addition, `localDev` will be set to true and `thirdPartyFrameHost` and `thirdPartyFrameRegex` will point to `AMP_TESTING_HOST` when the runtime is built and served from a non-localhost server, via the following commands.

```
gulp clean
export AMP_TESTING_HOST=<hostname.domain.com>
gulp build
gulp serve --host=<hostname.domain.com>
```

Partial fix for #12054

### CODE REVIEW

File: build-system/tasks/prepend-global/index.js [hunk 1]
```diff
@@ -50,13 +52,13 @@ function sanityCheck(str) {
 }
 
 /**
- * @param {string} filename
- * @param {string} opt_local
- * @param {string=} opt_branch
+ * @param {string} filename File containing the config
+ * @param {string} opt_localBranch Whether to use the local branch version
```
Comment (erwinmombay) [inline build-system/tasks/prepend-global/index.js]: if this is actually opt then add `=`

File: build-system/tasks/prepend-global/index.js [hunk 2]
```diff
@@ -110,46 +112,83 @@ function valueOrDefault(value, defaultValue) {
...
 /**
- * @param {string} config
- * @param {string} target
- * @param {string} filename
- * @param {string} opt_local
- * @param {string} opt_branch
+ * @param {string} config Prod or canary
+ * @param {string} target File containing the AMP runtime (amp.js or v0.js)
+ * @param {string} filename File containing the (prod or canary) config
+ * @param {boolean} opt_localDev Whether to enable local development
```
Comment (erwinmombay) [inline build-system/tasks/prepend-global/index.js]: missing `=` on these optional params

File: build-system/tasks/prepend-global/index.js (hunk 1 above)
Comment (rsimha-amp) [reply build-system/tasks/prepend-global/index.js]: Done.

File: build-system/tasks/prepend-global/index.js (hunk 2 above)
Comment (rsimha-amp) [reply build-system/tasks/prepend-global/index.js]: Done.

**Pull request merged**